riscv/simif.h

```diff
@@ -16,6 +16,10 @@ class simif_t
   virtual bool mmio_store(reg_t addr, size_t len, const uint8_t* bytes) = 0;
   // Callback for processors to let the simulation know they were reset.
   virtual void proc_reset(unsigned id) = 0;
+
+#ifdef RISCV_ENABLE_COMMITLOG
+  virtual const char* get_symbol(uint64_t addr) = 0;
+#endif
```

Comment:
I don't think we want to conditionalize the presence of a virtual function on a preprocessor macro; this will cause ABI issues when Spike is used as a library.  Can we just remove this entirely and have sim_t directly use the version defined in its superclass htif_t?

Reply:
The processor_t class only has access to the interface simif_t* and not to the sim_t object directly.
So I am not sure how to get hold of the sim_t object.

There are ways to do it, but all of them don't feel *clean*.
1. We could type cast the pointer to simif_t* in processor_t to sim_t*
2. Add a get_sim_t_obj()
3. Change virtual get_symbol(uint64_t addr) = 0; in simif.h to  virtual get_symbol(uint64_t addr) {nullptr};
    We have a check for nullptr, which indicates symbols not found.

Is there a cleaner way to get the sim_t* obj?

I have removed the unnecessary get_symbols() defs in sim.h/cpp, I will check it in once we have resolved the above issue.

Reply:
Ah, I forgot about that.  OK, yeah, it needs to go in simif_t.

Let's just keep the code as-is, but remove the `#ifdef`.  There's no harm in building this additional code when the commit log is disabled.  The extra time it takes to build the map at startup is tiny in the grand scheme of things.